modules/assim.sequential/tests/testthat/test_downscaling.R

```diff
@@ -0,0 +1,82 @@
+# Test setup: temporary datasets and files
+withr::with_tempdir({
+  temp_ensemble_data_rds <- "ensemble_data.rds"
+  temp_coords_csv <- "final_design_points.csv"
+  file.remove(temp_ensemble_data_rds, temp_coords_csv)
+
+  set.seed(123)
```

Comment:
Beware that the effects of this `set.seed` persist outside this test block! I think the easiest way to contain it would be to wrap all the lines that use random nunbers in one big `withr::with_seed(123, {...code that uses randomness here...})`.

(I think supposedly you could also replace this line with `withr::local_seed(123)` to have the seed reset at the end of the current execution context, but in my local tests that still affected code outside the function. Dunno if I'm misunderstanding "context" or I did something else wrong)